File: sdk-python/src/ort_vision_sdk/core/exceptions.py
"""Exceptions raised by the SDK.

All exceptions inherit from :class:`OrtVisionError`, so callers can catch the
base class to handle any SDK-originated failure uniformly.
"""

from __future__ import annotations


class OrtVisionError(Exception):
    """Base class for all ort-vision-sdk errors."""


class ModelLoadError(OrtVisionError):
    """Raised when an ONNX model cannot be loaded into an inference session."""


class InferenceError(OrtVisionError):
    """Raised when ONNX Runtime fails while executing a model."""


class ProviderNotAvailableError(OrtVisionError):
    """Raised when a requested execution provider is not available in this ORT build."""


class ImageLoadError(OrtVisionError):
    """Raised when an input image cannot be decoded into the canonical array format."""


class LabelMapError(OrtVisionError):
    """Raised when class labels cannot be resolved from the supplied spec."""


class FusionError(OrtVisionError):
    """Raised when two models cannot be fused into a single pipeline graph.

    Covers every way :mod:`ort_vision_sdk.compose` can refuse a pair of models:
    an unreadable file, a head layout the bridge cannot decode, opsets that
    cannot be reconciled, a classifier whose graph will not accept a batch of
    crops, or a fused graph that fails its post-build validation run.
    """

File: sdk-js-web/src/core/exceptions.ts
/**
 * Exceptions raised by the SDK.
 *
 * All exceptions inherit from {@link OrtVisionError}, so callers can catch
 * the base class to handle any SDK-originated failure uniformly.
 */

export class OrtVisionError extends Error {
  constructor(message: string, options?: ErrorOptions) {
    super(message, options);
    this.name = new.target.name;
  }
}

/** Raised when an ONNX model cannot be loaded into an inference session. */
export class ModelLoadError extends OrtVisionError {}

/** Raised when ONNX Runtime fails while executing a model. */
export class InferenceError extends OrtVisionError {}

/** Raised when a requested execution provider is not available. */
export class ProviderNotAvailableError extends OrtVisionError {}

/** Raised when an input image cannot be decoded into the canonical format. */
export class ImageLoadError extends OrtVisionError {}

/** Raised when class labels cannot be resolved from the supplied spec. */
export class LabelMapError extends OrtVisionError {}

/**
 * Raised when a model cannot be driven as a fused detect→classify pipeline.
 *
 * In the browser this means the file carries no `ovs.*` metadata — it is a
 * plain detector or classifier rather than something `ort_vision_sdk.compose`
 * produced — or the graph is missing an output the pipeline contract requires.
 * Building a pipeline is a Python-side build step; the browser only runs one.
 */
export class FusionError extends OrtVisionError {}
